src/meta/src/stream/stream_manager.rs

```diff
@@ -234,8 +247,10 @@ impl GlobalStreamManager {
                         .inspect_err(|_| tracing::warn!("failed to notify created: {table_id}"));
                 }
                 Err(err) => {
-                    for revert_func in revert_funcs.into_iter().rev() {
-                        revert_func.await;
+                    if create_type == CreateType::Foreground {
```

Comment:
Or can we initialize the revert functions according to its create type and leave the revert operation unchanged? It will make the revert logic less error-prone.